### PR TITLE
Implement Xprint-diff without external libraries.

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by odersky** [comment]
@nicolasstucki Can you propose a reviewer?

**Comment by nicolasstucki** [comment]
Review by @DarkDimius

**Comment by nicolasstucki** [comment]
BTW, I found some bugs with the current implementation. I did not try to minimize them and add them to the issue tracker because this PR fixes them.

File: src/dotty/tools/dotc/util/DiffUtil.scala [hunk 1]
```diff
@@ -30,38 +30,120 @@ object DiffUtil {
...
+    for (j <- y.indices)
+      F(0)(j) = d * j
+    for (i <- 1 to x.length) {
+      for (j <- 1 to y.length) {
+        val mtch = F(i - 1)(j - 1) + similarity(x(i-1), y(j-1))
+        val delete = F(i - 1)(j) + d
+        val insert = F(i)(j - 1) + d
+        F(i)(j) = mtch max insert max delete
+      }
+    }
```
Comment (julienrf) [inline src/dotty/tools/dotc/util/DiffUtil.scala]: Why not use `Array.tabulate` rather than `Array.fill`?

Comment (nicolasstucki) [inline src/dotty/tools/dotc/util/DiffUtil.scala]: It was to remain as close as possible to the pseudocode of the algorithm. I will change it now to save up two small traversals.

**Comment by odersky** [comment]
Can this go in now?

**Comment by nicolasstucki** [comment]
I should be ready to go in. In the last update I just changed a couple of variable names that Codacy suggested and I completely agreed with it in those cases.

File: src/dotty/tools/dotc/util/DiffUtil.scala (hunk 1 above)
Comment (nicolasstucki) [inline src/dotty/tools/dotc/util/DiffUtil.scala]: Implemented

**Comment by DarkDimius** [comment]
LGTM

**Pull request merged**